tutorials/plugins/running_code_in_the_editor.rst

```diff
@@ -286,6 +286,89 @@ By default, the warning only updates when closing and reopening the scene.
+------------------------------------------
+
+Sometimes, you need to run code just one time to automate a certain task that is
+not available in the editor out of the box. Some examples might be:
+
+- Scale all light nodes in the currently edited scene, as you noticed your level
+  ends up looking too dark or too bright after placing lights where desired.
+- Replace nodes that were copy-pasted with scene instances to make them easier
+  to modify later.
+
+This is available in Godot by extending :ref:`class_EditorScript` in a script.
+This provides a way to run individual scripts in the editor without having to
+create an editor plugin.
+
+To create an EditorScript, right-click a folder or empty space in the FileSystem
+dock then choose **New > Script...**. In the script creation dialog, click the
+tree icon to choose an object to extend from (or enter ``EditorScript`` directly
+in the field on the left, though note this is case-sensitive):
+
+.. figure:: img/running_code_in_the_editor_creating_editor_script.webp
+   :align: center
+   :alt: Creating an editor script in the script editor creation dialog
+
+   Creating an editor script in the script editor creation dialog
+
+This will automatically select a script template that is suited for
+EditorScripts, with a ``_run()`` method already inserted:
+
+::
+
+    @tool
+    extends EditorScript
+
+    # Called when the script is executed (using File -> Run in Script Editor).
+    func _run() -> void:
+        pass
+
+This ``_run()`` method is executed when you use **File > Run** or the keyboard
+shortcut :kbd:`Ctrl + Shift + X` while the EditorScript is the currently open
+script in the script editor.
+
+Scripts that extend EditorScript must be ``@tool`` scripts to function.
+
+.. warning::
+
+    EditorScripts have no undo/redo functionality, so **make sure to save your
+    scene before running one** if the script is designed to modify any data.
+
+To access nodes in the currently edited scene, use the
+:ref:`EditorScript.get_scene <class_EditorScript_method_get_scene>` method which
+returns the root Node of the currently edited scene. Here's an example that
+recursively gets all nodes in the currently edited scene and doubles the range
+of all OmniLight3D nodes:
+
+::
+
+    @tool
+    extends EditorScript
+
+    func _run() -> void:
+        for node in get_all_children(get_scene()):
+            if node is OmniLight3D:
+                node.omni_range *= 2.0
+
+    # This function is recursive: it calls itself to get lower levels of child nodes as needed.
+    # `children_acc` is the accumulator parameter that allows this function to work.
+    # It should be left to its default value when you call this function directly.
+    func get_all_children(in_node, children_acc = []):
+        children_acc.push_back(in_node)
+        for child in in_node.get_children():
+            children_acc = get_all_children(child, children_acc)
+
+        return children_acc
```

Comment:
I'd like this script not to affect children of instanced subscenes, but I'm not sure how to detect those (regardless of whether Editable Children is enabled or not on them).